Gajeongtongsin/Gajeongtongsin/Screens/Parent/Reservation/ParentsCalenderViewController.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  ReservationViewController.swift
+//  Gajeongtongsin
+//
+//  Created by DaeSeong on 2022/07/16.
+//
+
+import UIKit
+
+class ParentsCalenderViewController: BaseViewController {
+    
+    //MARK: - Properties
+    private var choicedCells: [Bool] = Array(repeating: false, count:30) //복수선택 및 선택취소를 위한 array
+    private var subIdx: [Int] = [] //신청버튼 클릭 후 신청내역 인덱스가 저장되는 리스트
+    private var subDate: [String] = []
+
+    // 캘린더뷰
+    private let calenderView:  UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.register(CalenderViewCell.self, forCellWithReuseIdentifier: CalenderViewCell.identifier)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false //필수 !!
+        return collectionView
+    }()
+
+    
+    // 신청버튼
+    private let subBtn: UIButton = {
+        let button = UIButton()
+        button.setTitle("신청하기", for: .normal)
+        button.setTitleColor(.black, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        calenderView.delegate = self
+        calenderView.dataSource = self
+        
+        subBtn.addTarget(self, action: #selector(onTapButton), for: .touchUpInside)
+    }
+    
+    //날자 계산을 위한 상수 (근데 왜 let 선언하면 오류가 뜰까?)
+    var interval: Double {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MMM-dd-e-EEEE"    //e는 1~7(sun~sat)
+
+        let day = formatter.string(from:Date())
+        let today = day.components(separatedBy: "-")
+        let interval = Double(today[2])
+//        let startDay = Date(timeIntervalSinceNow: (86400 * (9-interval!))) //e: 1~7 이므로 7+2(월요일 보정)-interval = 다음주 월요일
+//        let startDayString = formatter.string(from: startDay).components(separatedBy: "-") //다음주 월요일 date
+        return interval!
+    }
+    
+    //신청하기 누르면 리로드 & 신청요일, 시간 mackdata에 추가 / print
+    @objc func onTapButton() {
```

Comment:
P1;
탭버튼에 너무 많은 기능들이 많아서 레브도 가독성이 떨어진다고 하신 걸 고려해서
기능별로 함수로 빼서 함수를 호출하는 방식으로 하면 가독성이 훨씬 좋아질 거 같아요.
아래는 대략적인 예시이고, 레브의 입맛에 맞게 세분화하면 좋을 거같습니다!
```suggestion
  @objc func onTapButton() {
        var subSchedule: [ScheduleInfo] = []
        
        subIdx = choicedCells.enumerated().compactMap { (idx, element) -> Int? in
            element ? idx : nil
        }
        
        for idx in subIdx {
            subSchedule.append(ScheduleInfo(consultingDate: getconsultingDate(idx: idx),
                                            startTime: getStartTime(idx: idx),
                                            isReserved: nil))
        }
        
        let newSchedule = Schedule(
            reservedDate: "7월22일",
            scheduleList: subSchedule,
            content: "테스트")
        
        parentList[0].schedules.append(newSchedule)
        
        print(parentList[0].schedules[1])
        choicedCells = Array(repeating: false, count:30)
        calenderView.reloadData()
    }```

Reply:
수정했습니다 ! 훨씬 깔끔해졌네요 고맙습니다